pkg/apis/core/v1alpha1/conversions.go

```diff
@@ -1549,6 +1551,31 @@ func Convert_garden_Shoot_To_v1alpha1_Shoot(in *garden.Shoot, out *Shoot, s conv
 	out.Spec.SecretBindingName = in.Spec.Cloud.SecretBindingRef.Name
 	out.Spec.SeedName = in.Spec.Cloud.Seed
 
+	if email, ok := in.Annotations[constants.AnnotationShootOperatedBy]; ok && utils.TestEmail(email) {
+		exists := false
+		if in.Spec.Monitoring == nil {
+			out.Spec.Monitoring = &Monitoring{
+				Alerting: &Alerting{},
+			}
+		}
+		if in.Spec.Monitoring != nil && in.Spec.Monitoring.Alerting == nil {
+			out.Spec.Monitoring.Alerting = &Alerting{}
+		}
+		if in.Spec.Monitoring != nil && in.Spec.Monitoring.Alerting != nil {
+			for _, receiver := range in.Spec.Monitoring.Alerting.EmailReceivers {
+				if receiver == email {
+					exists = true
+					break
+				}
+			}
+		}
```

Comment:
Make it better readable:

```suggestion
		if in.Spec.Monitoring == nil {
			out.Spec.Monitoring = &Monitoring{}
		}
		if in.Spec.Monitoring.Alerting == nil {
			out.Spec.Monitoring.Alerting = &Alerting{}
		}
		for _, receiver := range in.Spec.Monitoring.Alerting.EmailReceivers {
			...
```

Reply:
This won't work if `in.Spec.Monitoring` is nil because when it checks `in.Spec.Monitoring.Alerting` Monitoring is still nil. Am I missing something?

Reply:
I don't understand. If `in.spec.Monitoring == nil` then it will be set to `&Monitoring{}` in line `1572`. Hence, the check for `in.Spec.Monitoring.Alerting == nil` in line `1574` can savely assume that `Monitoring != nil`.

Reply:
We are checking in.Spec... but setting out.Spec...

Reply:
OK, I see... :) thanks, then let's get it in